src/shared/schema/avl.schema.ts

```diff
@@ -241,33 +242,42 @@ export const tflVehicleLocationSchemaTransformed = tflVehicleLocationSchema.tran
         item_id: item.item_id,
         valid_until_time: validUntilTime,
         producer_ref: item.producerRef,
-        vehicle_ref: item.vehicleRef,
+        vehicle_ref: item.vehicleRef?.replaceAll(NM_TOKEN_DISALLOWED_CHARS_REGEX, ""),
         vehicle_name: item.vehicleName,
-        operator_ref: item.operatorRef,
+        operator_ref: item.operatorRef?.replaceAll(NM_TOKEN_DISALLOWED_CHARS_REGEX, ""),
         monitored: item.monitored,
         longitude: item.longitude,
         latitude: item.latitude,
         recorded_at_time: recordedAtTime,
-        bearing: typeof item.bearing === "number" ? item.bearing.toString() : item.bearing,
+        bearing: item.bearing?.toString(),
         load: item.load,
         passenger_count: item.passengerCount,
         odometer: item.odometer,
         headway_deviation: item.headwayDeviation,
         schedule_deviation: item.scheduleDeviation,
         vehicle_state: item.vehicleState,
         next_stop_point_id: item.nextStopPointId,
-        next_stop_point_name: item.nextStopPointName,
+        next_stop_point_name: item.nextStopPointName?.replaceAll(
+            SIRI_VM_POPULATED_STRING_TYPE_DISALLOWED_CHARS_REGEX,
+            "",
+        ),
         previous_stop_point_id: item.previousStopPointId,
-        previous_stop_point_name: item.previousStopPointName,
-        line_ref: item.lineRef,
-        published_line_name: item.publishedLineName,
+        previous_stop_point_name: item.previousStopPointName?.replaceAll(
+            SIRI_VM_POPULATED_STRING_TYPE_DISALLOWED_CHARS_REGEX,
+            "",
+        ),
+        line_ref: item.lineRef?.replaceAll(NM_TOKEN_DISALLOWED_CHARS_REGEX, ""),
+        published_line_name: item.publishedLineName?.replaceAll(
+            SIRI_VM_POPULATED_STRING_TYPE_DISALLOWED_CHARS_REGEX,
+            "",
+        ),
         direction_ref: item.directionRef.toString(),
```

Comment:
should directionRef have validation here - looks like it did on SIRI generation

Reply:
directionRef comes as a number from TfL so validation isn't needed here